sql/planbuilder/from.go

```diff
@@ -748,6 +748,16 @@ func (b *Builder) resolveView(name string, database sql.Database, asOf interface
 			b.parserOpts = sql.NewSqlModeFromString(viewDef.SqlMode).ParserOptions()
 			node, _, _, err := b.Parse(viewDef.TextDefinition, false)
 			if err != nil {
+				// TODO: Need to account for non-existing functions or
+				//  users without appropriate privilege to the referenced table/column/function.
+				if sql.ErrTableNotFound.Is(err) || sql.ErrColumnNotFound.Is(err) {
+					vd, exists, _ := vdb.GetViewDefinition(b.ctx, name)
+					// whether the view exists will eliminate CREATE VIEW stmts
+					// whether the view TextDefinition matches will eliminate ALTER VIEW stmts
+					if exists && vd.TextDefinition == viewDef.TextDefinition {
+						err = sql.ErrInvalidRefInView.New(database.Name(), name)
+					}
+				}
```

Comment:
Nice! This looks a lot better. I'm not sure you need the call to `GetViewDefinition` on line 754 though... it seems like line 731 has already called `GetViewDefinition` and loaded the view definition for `name`, so we know that it always already exists if we get to line 754. 

Since we don't support `alter view` yet, I'd be okay with just leaving a TODO comment in this block about how ALTER VIEW should not return this error, and not bothering to check the `TextDefinition` fields.